### ISSUE TITLE
linter: func-style rule missing allowArrowFunctions option

### ISSUE THREAD

**Issue by yardenhochman** [oxc-project/oxc#20618]
### What version of Oxlint are you using?

1.41.0

### What command did you run?

```bash
npx oxlint -c .oxlintrc.json src/shared/file-utils.ts
```

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "rules": {
    "func-style": ["error", "declaration", { "allowArrowFunctions": true }]
  }
}
```

### What happened?

The `allowArrowFunctions` option is ignored. Arrow function expressions assigned to variables are still flagged even when `allowArrowFunctions: true` is set.

### What did you expect to happen?

With `allowArrowFunctions: true`, the rule should only flag `const foo = function() {}` (function expressions) but allow `const foo = () => {}` (arrow functions), matching the [ESLint behavior](https://eslint.org/docs/latest/rules/func-style#allowarrowfunctions).

This option is useful for codebases that prefer function declarations but need arrow expressions for certain patterns (e.g., React components with HOCs, callbacks assigned to variables).

**Comment by huangtiandi1999** [comment]
I can't reproduce this issue. Tested with v1.59 and the behavior is consistent with ESLint.

**Comment by camc314** [comment]
Hi, I tried to reproduce this issue here: https://github.com/camc314/oxlint-repros/commit/0d3cf793b03dfc27658028c7645527b8776fbc38

But I observed the same as huangtiandi1999 - both oxlint and eslint have the same behaviour.

Please clarify the issue you're observing

Thanks
